File: src/test/java/com/roscopeco/jasm/model/LdcAconstAreturn.java
package com.roscopeco.jasm.model;

import java.lang.invoke.MethodHandle;
import java.lang.invoke.MethodType;

public interface LdcAconstAreturn {
    Object testAconstNull();

    String testLdcString();

    int testLdcInt();

    float testLdcFloat();

    boolean testLdcBool();

    Class<?> testLdcClass();

    MethodType testLdcMethodType();

    MethodHandle testLdcMethodHandle();

    String testLdcDynamicConst();
}

File: src/test/kotlin/com/roscopeco/jasm/model/LdcAconstAreturn.kt
package com.roscopeco.jasm.model

import java.lang.invoke.MethodHandle
import java.lang.invoke.MethodType

interface LdcAconstAreturn {
    fun testAconstNull(): Any?
    fun testLdcString(): String
    fun testLdcInt(): Int
    fun testLdcFloat(): Float
    fun testLdcLong(): Long
    fun testLdcDouble(): Double
    fun testLdcBool(): Boolean
    fun testLdcClass(): Class<*>
    fun testLdcMethodType(): MethodType
    fun testLdcMethodHandle(): MethodHandle
    fun testLdcDynamicConst(): String
}